### ISSUE TITLE
Failing tests: 3.0.5

### ISSUE THREAD

**Issue by sanjayankur31** [jonathf/chaospy#160]
Hello, with 3.0.5, we see some tests failing:

```
+ pytest-3.7 tests
============================= test session starts ==============================
platform linux -- Python 3.7.4, pytest-4.6.4, py-1.8.0, pluggy-0.12.0
rootdir: /builddir/build/BUILD/chaospy-3.0.5, inifile: setup.cfg
plugins: cov-2.7.1
collected 189 items
tests/test_dist.py ..................................................... [ 28%]
........................................................................ [ 66%]
.....................................                                    [ 85%]
tests/test_orth.py .....                                                 [ 88%]
tests/test_poly.py ....FF..F.....                                        [ 95%]
tests/test_stress.py ...F....                                            [100%]
=================================== FAILURES ===================================
_______________________________ test_poly_basis ________________________________
    def test_poly_basis():
        basis = cp.basis(1, 2, 2)
>       assert str(basis) == "[q0, q1, q0^2, q0q1, q1^2]"
E       AssertionError: assert '[q1, q0, q1^2, q0q1, q0^2]' == '[q0, q1, q0^2, q0q1, q1^2]'
E         - [q1, q0, q1^2, q0q1, q0^2]
E         ?        ----           ^
E         + [q0, q1, q0^2, q0q1, q1^2]
E         ?  ++++                 ^
/builddir/build/BUILD/chaospy-3.0.5/tests/test_poly.py:32: AssertionError
______________________________ test_poly_dimredux ______________________________
    def test_poly_dimredux():
        basis = cp.basis(1, 2, 2)
        X = cp.variable()
>       assert basis[0] == X
E       assert <chaospy.poly...x7fff87b0d510> == <chaospy.poly....x7fff87b0d0d0>
E         (pytest_assertion plugin: representation of details failed.  Probably an object has a faulty __repr__.)
E         /builddir/build/BUILD/chaospy-3.0.5/chaospy/poly/base.py:239: ValueError: not iterable
/builddir/build/BUILD/chaospy-3.0.5/tests/test_poly.py:38: AssertionError
___________________________ test_poly_indirect_evals ___________________________
    def test_poly_indirect_evals():
        basis = cp.basis(1, 2, 2)
        x, y = cp.variable(2)
>       assert basis(q1=3)[1] == 3
E       assert <chaospy.poly...x7fff879dad10> == 3
E         (pytest_assertion plugin: representation of details failed.  Probably an object has a faulty __repr__.)
E         /builddir/build/BUILD/chaospy-3.0.5/chaospy/poly/base.py:239: ValueError: not iterable
/builddir/build/BUILD/chaospy-3.0.5/tests/test_poly.py:56: AssertionError
_______________________________ test_quadrature ________________________________
    def test_quadrature():
        dist = cp.Iid(cp.Normal(), dim)
        gq = cp.generate_quadrature
        nodes, weights = gq(order, dist, rule="C")
        nodes, weights = gq(order, dist, rule="E")
        nodes, weights = gq(order, dist, rule="G")
>       nodes, weights = gq(order, dist, rule="C", sparse=True)
/builddir/build/BUILD/chaospy-3.0.5/tests/test_stress.py:67: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/builddir/build/BUILD/chaospy-3.0.5/chaospy/quad/interface.py:71: in generate_quadrature
    abscissas, weights = sparse_grid.sparse_grid(quad_function, order, dim)
/builddir/build/BUILD/chaospy-3.0.5/chaospy/quad/sparse_grid.py:73: in sparse_grid
    return sparse_grid(func, m_order, dim, skew)
/builddir/build/BUILD/chaospy-3.0.5/chaospy/quad/sparse_grid.py:86: in sparse_grid
    - chaospy.bertran.terms(order-dim, dim)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
order = -1, dim = 3
    def terms(order, dim):
        """
        Count the number of polynomials in an expansion.
    
        Args:
            order (int):
                The upper order for the expansion.
            dim (int):
                The number of dimensions of the expansion.
    
        Returns:
            The number of terms in an expansion of upper order ``order`` and number
            of dimensions ``dim``.
        """
        return int(math.factorial(order+dim)/
>                  (math.factorial(order)*math.factorial(dim)))
E       ValueError: factorial() not defined for negative values
/builddir/build/BUILD/chaospy-3.0.5/chaospy/bertran/operators.py:62: ValueError
=============================== warnings summary ===============================
chaospy/poly/__init__.py:95
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/poly/__init__.py:95: DeprecationWarning: invalid escape sequence \_
    """
chaospy/distributions/sampler/__init__.py:22
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/sampler/__init__.py:22: DeprecationWarning: invalid escape sequence \s
    """
chaospy/distributions/copulas/baseclass.py:62
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/copulas/baseclass.py:62: DeprecationWarning: invalid escape sequence \i
    """
tests/test_dist.py::test_dist_add[Alpha]
tests/test_dist.py::test_dist_add[Alpha]
tests/test_dist.py::test_dist_add[Alpha]
tests/test_dist.py::test_dist_add[Anglit]
tests/test_dist.py::test_dist_add[Anglit]
tests/test_dist.py::test_dist_add[Anglit]
tests/test_dist.py::test_dist_add[Bradford]
tests/test_dist.py::test_dist_add[Bradford]
tests/test_dist.py::test_dist_add[Bradford]
tests/test_dist.py::test_dist_add[Cauchy]
tests/test_dist.py::test_dist_add[Cauchy]
tests/test_dist.py::test_dist_add[Cauchy]
tests/test_dist.py::test_dist_add[ChiSquared]
tests/test_dist.py::test_dist_add[ChiSquared]
tests/test_dist.py::test_dist_add[ChiSquared]
tests/test_dist.py::test_dist_add[DoubleGamma]
tests/test_dist.py::test_dist_add[DoubleGamma]
tests/test_dist.py::test_dist_add[DoubleGamma]
tests/test_dist.py::test_dist_add[DoubleWeibull]
tests/test_dist.py::test_dist_add[DoubleWeibull]
tests/test_dist.py::test_dist_add[DoubleWeibull]
tests/test_dist.py::test_dist_add[ExponentialPower]
tests/test_dist.py::test_dist_add[ExponentialPower]
tests/test_dist.py::test_dist_add[ExponentialPower]
tests/test_dist.py::test_dist_add[ExponentialWeibull]
tests/test_dist.py::test_dist_add[ExponentialWeibull]
tests/test_dist.py::test_dist_add[ExponentialWeibull]
tests/test_dist.py::test_dist_add[F]
tests/test_dist.py::test_dist_add[F]
tests/test_dist.py::test_dist_add[F]
tests/test_dist.py::test_dist_add[FatigueLife]
tests/test_dist.py::test_dist_add[FatigueLife]
tests/test_dist.py::test_dist_add[FatigueLife]
tests/test_dist.py::test_dist_add[Fisk]
tests/test_dist.py::test_dist_add[Fisk]
tests/test_dist.py::test_dist_add[Fisk]
tests/test_dist.py::test_dist_add[FoldedCauchy]
tests/test_dist.py::test_dist_add[FoldedCauchy]
tests/test_dist.py::test_dist_add[FoldedCauchy]
tests/test_dist.py::test_dist_add[FoldedNormal]
tests/test_dist.py::test_dist_add[FoldedNormal]
tests/test_dist.py::test_dist_add[FoldedNormal]
tests/test_dist.py::test_dist_add[GeneralizedExponential]
tests/test_dist.py::test_dist_add[GeneralizedExponential]
tests/test_dist.py::test_dist_add[GeneralizedExponential]
tests/test_dist.py::test_dist_add[GeneralizedExtreme]
tests/test_dist.py::test_dist_add[GeneralizedExtreme]
tests/test_dist.py::test_dist_add[GeneralizedExtreme]
tests/test_dist.py::test_dist_add[Levy]
tests/test_dist.py::test_dist_add[Levy]
tests/test_dist.py::test_dist_add[Levy]
tests/test_dist.py::test_dist_add[LogGamma]
tests/test_dist.py::test_dist_add[LogGamma]
tests/test_dist.py::test_dist_add[LogGamma]
tests/test_dist.py::test_dist_add[LogLaplace]
tests/test_dist.py::test_dist_add[LogLaplace]
tests/test_dist.py::test_dist_add[LogLaplace]
tests/test_dist.py::test_dist_add[LogWeibull]
tests/test_dist.py::test_dist_add[LogWeibull]
tests/test_dist.py::test_dist_add[LogWeibull]
tests/test_dist.py::test_dist_add[Logistic]
tests/test_dist.py::test_dist_add[Logistic]
tests/test_dist.py::test_dist_add[Logistic]
tests/test_dist.py::test_dist_add[Mielke]
tests/test_dist.py::test_dist_add[Mielke]
tests/test_dist.py::test_dist_add[Mielke]
tests/test_dist.py::test_dist_add[Nakagami]
tests/test_dist.py::test_dist_add[Nakagami]
tests/test_dist.py::test_dist_add[Nakagami]
tests/test_dist.py::test_dist_add[Pareto1]
tests/test_dist.py::test_dist_add[Pareto1]
tests/test_dist.py::test_dist_add[Pareto1]
tests/test_dist.py::test_dist_add[Pareto2]
tests/test_dist.py::test_dist_add[Pareto2]
tests/test_dist.py::test_dist_add[Pareto2]
tests/test_dist.py::test_dist_add[PowerLogNormal]
tests/test_dist.py::test_dist_add[PowerLogNormal]
tests/test_dist.py::test_dist_add[PowerLogNormal]
tests/test_dist.py::test_dist_add[PowerNormal]
tests/test_dist.py::test_dist_add[PowerNormal]
tests/test_dist.py::test_dist_add[PowerNormal]
tests/test_dist.py::test_dist_add[TruncExponential]
tests/test_dist.py::test_dist_add[TruncExponential]
tests/test_dist.py::test_dist_add[TruncExponential]
tests/test_dist.py::test_dist_add[TruncNormal]
tests/test_dist.py::test_dist_add[TruncNormal]
tests/test_dist.py::test_dist_add[TruncNormal]
tests/test_dist.py::test_dist_add[TukeyLambda]
tests/test_dist.py::test_dist_add[TukeyLambda]
tests/test_dist.py::test_dist_add[TukeyLambda]
tests/test_dist.py::test_dist_add[Wald]
tests/test_dist.py::test_dist_add[Wald]
tests/test_dist.py::test_dist_add[Wald]
tests/test_dist.py::test_dist_add[WrappedCauchy]
tests/test_dist.py::test_dist_add[WrappedCauchy]
tests/test_dist.py::test_dist_add[WrappedCauchy]
tests/test_dist.py::test_dist_sub[Alpha]
tests/test_dist.py::test_dist_sub[Alpha]
tests/test_dist.py::test_dist_sub[Alpha]
tests/test_dist.py::test_dist_sub[Anglit]
tests/test_dist.py::test_dist_sub[Anglit]
tests/test_dist.py::test_dist_sub[Anglit]
tests/test_dist.py::test_dist_sub[Bradford]
tests/test_dist.py::test_dist_sub[Bradford]
tests/test_dist.py::test_dist_sub[Bradford]
tests/test_dist.py::test_dist_sub[Cauchy]
tests/test_dist.py::test_dist_sub[Cauchy]
tests/test_dist.py::test_dist_sub[Cauchy]
tests/test_dist.py::test_dist_sub[ChiSquared]
tests/test_dist.py::test_dist_sub[ChiSquared]
tests/test_dist.py::test_dist_sub[ChiSquared]
tests/test_dist.py::test_dist_sub[DoubleGamma]
tests/test_dist.py::test_dist_sub[DoubleGamma]
tests/test_dist.py::test_dist_sub[DoubleGamma]
tests/test_dist.py::test_dist_sub[DoubleWeibull]
tests/test_dist.py::test_dist_sub[DoubleWeibull]
tests/test_dist.py::test_dist_sub[DoubleWeibull]
tests/test_dist.py::test_dist_sub[ExponentialPower]
tests/test_dist.py::test_dist_sub[ExponentialPower]
tests/test_dist.py::test_dist_sub[ExponentialPower]
tests/test_dist.py::test_dist_sub[ExponentialWeibull]
tests/test_dist.py::test_dist_sub[ExponentialWeibull]
tests/test_dist.py::test_dist_sub[ExponentialWeibull]
tests/test_dist.py::test_dist_sub[F]
tests/test_dist.py::test_dist_sub[F]
tests/test_dist.py::test_dist_sub[F]
tests/test_dist.py::test_dist_sub[FatigueLife]
tests/test_dist.py::test_dist_sub[FatigueLife]
tests/test_dist.py::test_dist_sub[FatigueLife]
tests/test_dist.py::test_dist_sub[Fisk]
tests/test_dist.py::test_dist_sub[Fisk]
tests/test_dist.py::test_dist_sub[Fisk]
tests/test_dist.py::test_dist_sub[FoldedCauchy]
tests/test_dist.py::test_dist_sub[FoldedCauchy]
tests/test_dist.py::test_dist_sub[FoldedCauchy]
tests/test_dist.py::test_dist_sub[FoldedNormal]
tests/test_dist.py::test_dist_sub[FoldedNormal]
tests/test_dist.py::test_dist_sub[FoldedNormal]
tests/test_dist.py::test_dist_sub[GeneralizedExponential]
tests/test_dist.py::test_dist_sub[GeneralizedExponential]
tests/test_dist.py::test_dist_sub[GeneralizedExponential]
tests/test_dist.py::test_dist_sub[GeneralizedExtreme]
tests/test_dist.py::test_dist_sub[GeneralizedExtreme]
tests/test_dist.py::test_dist_sub[GeneralizedExtreme]
tests/test_dist.py::test_dist_sub[Levy]
tests/test_dist.py::test_dist_sub[Levy]
tests/test_dist.py::test_dist_sub[Levy]
tests/test_dist.py::test_dist_sub[LogGamma]
tests/test_dist.py::test_dist_sub[LogGamma]
tests/test_dist.py::test_dist_sub[LogGamma]
tests/test_dist.py::test_dist_sub[LogLaplace]
tests/test_dist.py::test_dist_sub[LogLaplace]
tests/test_dist.py::test_dist_sub[LogLaplace]
tests/test_dist.py::test_dist_sub[LogWeibull]
tests/test_dist.py::test_dist_sub[LogWeibull]
tests/test_dist.py::test_dist_sub[LogWeibull]
tests/test_dist.py::test_dist_sub[Logistic]
tests/test_dist.py::test_dist_sub[Logistic]
tests/test_dist.py::test_dist_sub[Logistic]
tests/test_dist.py::test_dist_sub[Mielke]
tests/test_dist.py::test_dist_sub[Mielke]
tests/test_dist.py::test_dist_sub[Mielke]
tests/test_dist.py::test_dist_sub[Nakagami]
tests/test_dist.py::test_dist_sub[Nakagami]
tests/test_dist.py::test_dist_sub[Nakagami]
tests/test_dist.py::test_dist_sub[Pareto1]
tests/test_dist.py::test_dist_sub[Pareto1]
tests/test_dist.py::test_dist_sub[Pareto1]
tests/test_dist.py::test_dist_sub[Pareto2]
tests/test_dist.py::test_dist_sub[Pareto2]
tests/test_dist.py::test_dist_sub[Pareto2]
tests/test_dist.py::test_dist_sub[PowerLogNormal]
tests/test_dist.py::test_dist_sub[PowerLogNormal]
tests/test_dist.py::test_dist_sub[PowerLogNormal]
tests/test_dist.py::test_dist_sub[PowerNormal]
tests/test_dist.py::test_dist_sub[PowerNormal]
tests/test_dist.py::test_dist_sub[PowerNormal]
tests/test_dist.py::test_dist_sub[TruncExponential]
tests/test_dist.py::test_dist_sub[TruncExponential]
tests/test_dist.py::test_dist_sub[TruncExponential]
tests/test_dist.py::test_dist_sub[TruncNormal]
tests/test_dist.py::test_dist_sub[TruncNormal]
tests/test_dist.py::test_dist_sub[TruncNormal]
tests/test_dist.py::test_dist_sub[TukeyLambda]
tests/test_dist.py::test_dist_sub[TukeyLambda]
tests/test_dist.py::test_dist_sub[TukeyLambda]
tests/test_dist.py::test_dist_sub[Wald]
tests/test_dist.py::test_dist_sub[Wald]
tests/test_dist.py::test_dist_sub[Wald]
tests/test_dist.py::test_dist_sub[WrappedCauchy]
tests/test_dist.py::test_dist_sub[WrappedCauchy]
tests/test_dist.py::test_dist_sub[WrappedCauchy]
tests/test_dist.py::test_dist_mul[Alpha]
tests/test_dist.py::test_dist_mul[Alpha]
tests/test_dist.py::test_dist_mul[Alpha]
tests/test_dist.py::test_dist_mul[Alpha]
tests/test_dist.py::test_dist_mul[Alpha]
tests/test_dist.py::test_dist_mul[Alpha]
tests/test_dist.py::test_dist_mul[Anglit]
tests/test_dist.py::test_dist_mul[Anglit]
tests/test_dist.py::test_dist_mul[Anglit]
tests/test_dist.py::test_dist_mul[Anglit]
tests/test_dist.py::test_dist_mul[Anglit]
tests/test_dist.py::test_dist_mul[Anglit]
tests/test_dist.py::test_dist_mul[Bradford]
tests/test_dist.py::test_dist_mul[Bradford]
tests/test_dist.py::test_dist_mul[Bradford]
tests/test_dist.py::test_dist_mul[Bradford]
tests/test_dist.py::test_dist_mul[Bradford]
tests/test_dist.py::test_dist_mul[Bradford]
tests/test_dist.py::test_dist_mul[Cauchy]
tests/test_dist.py::test_dist_mul[Cauchy]
tests/test_dist.py::test_dist_mul[Cauchy]
tests/test_dist.py::test_dist_mul[Cauchy]
tests/test_dist.py::test_dist_mul[Cauchy]
tests/test_dist.py::test_dist_mul[Cauchy]
tests/test_dist.py::test_dist_mul[ChiSquared]
tests/test_dist.py::test_dist_mul[ChiSquared]
tests/test_dist.py::test_dist_mul[ChiSquared]
tests/test_dist.py::test_dist_mul[ChiSquared]
tests/test_dist.py::test_dist_mul[ChiSquared]
tests/test_dist.py::test_dist_mul[ChiSquared]
tests/test_dist.py::test_dist_mul[DoubleGamma]
tests/test_dist.py::test_dist_mul[DoubleGamma]
tests/test_dist.py::test_dist_mul[DoubleGamma]
tests/test_dist.py::test_dist_mul[DoubleGamma]
tests/test_dist.py::test_dist_mul[DoubleGamma]
tests/test_dist.py::test_dist_mul[DoubleGamma]
tests/test_dist.py::test_dist_mul[DoubleWeibull]
tests/test_dist.py::test_dist_mul[DoubleWeibull]
tests/test_dist.py::test_dist_mul[DoubleWeibull]
tests/test_dist.py::test_dist_mul[DoubleWeibull]
tests/test_dist.py::test_dist_mul[DoubleWeibull]
tests/test_dist.py::test_dist_mul[DoubleWeibull]
tests/test_dist.py::test_dist_mul[ExponentialPower]
tests/test_dist.py::test_dist_mul[ExponentialPower]
tests/test_dist.py::test_dist_mul[ExponentialPower]
tests/test_dist.py::test_dist_mul[ExponentialPower]
tests/test_dist.py::test_dist_mul[ExponentialPower]
tests/test_dist.py::test_dist_mul[ExponentialPower]
tests/test_dist.py::test_dist_mul[ExponentialWeibull]
tests/test_dist.py::test_dist_mul[ExponentialWeibull]
tests/test_dist.py::test_dist_mul[ExponentialWeibull]
tests/test_dist.py::test_dist_mul[ExponentialWeibull]
tests/test_dist.py::test_dist_mul[ExponentialWeibull]
tests/test_dist.py::test_dist_mul[ExponentialWeibull]
tests/test_dist.py::test_dist_mul[F]
tests/test_dist.py::test_dist_mul[F]
tests/test_dist.py::test_dist_mul[F]
tests/test_dist.py::test_dist_mul[F]
tests/test_dist.py::test_dist_mul[F]
tests/test_dist.py::test_dist_mul[F]
tests/test_dist.py::test_dist_mul[FatigueLife]
tests/test_dist.py::test_dist_mul[FatigueLife]
tests/test_dist.py::test_dist_mul[FatigueLife]
tests/test_dist.py::test_dist_mul[FatigueLife]
tests/test_dist.py::test_dist_mul[FatigueLife]
tests/test_dist.py::test_dist_mul[FatigueLife]
tests/test_dist.py::test_dist_mul[Fisk]
tests/test_dist.py::test_dist_mul[Fisk]
tests/test_dist.py::test_dist_mul[Fisk]
tests/test_dist.py::test_dist_mul[Fisk]
tests/test_dist.py::test_dist_mul[Fisk]
tests/test_dist.py::test_dist_mul[Fisk]
tests/test_dist.py::test_dist_mul[FoldedCauchy]
tests/test_dist.py::test_dist_mul[FoldedCauchy]
tests/test_dist.py::test_dist_mul[FoldedCauchy]
tests/test_dist.py::test_dist_mul[FoldedCauchy]
tests/test_dist.py::test_dist_mul[FoldedCauchy]
tests/test_dist.py::test_dist_mul[FoldedCauchy]
tests/test_dist.py::test_dist_mul[FoldedNormal]
tests/test_dist.py::test_dist_mul[FoldedNormal]
tests/test_dist.py::test_dist_mul[FoldedNormal]
tests/test_dist.py::test_dist_mul[FoldedNormal]
tests/test_dist.py::test_dist_mul[FoldedNormal]
tests/test_dist.py::test_dist_mul[FoldedNormal]
tests/test_dist.py::test_dist_mul[GeneralizedExponential]
tests/test_dist.py::test_dist_mul[GeneralizedExponential]
tests/test_dist.py::test_dist_mul[GeneralizedExponential]
tests/test_dist.py::test_dist_mul[GeneralizedExponential]
tests/test_dist.py::test_dist_mul[GeneralizedExponential]
tests/test_dist.py::test_dist_mul[GeneralizedExponential]
tests/test_dist.py::test_dist_mul[GeneralizedExtreme]
tests/test_dist.py::test_dist_mul[GeneralizedExtreme]
tests/test_dist.py::test_dist_mul[GeneralizedExtreme]
tests/test_dist.py::test_dist_mul[GeneralizedExtreme]
tests/test_dist.py::test_dist_mul[GeneralizedExtreme]
tests/test_dist.py::test_dist_mul[GeneralizedExtreme]
tests/test_dist.py::test_dist_mul[Levy]
tests/test_dist.py::test_dist_mul[Levy]
tests/test_dist.py::test_dist_mul[Levy]
tests/test_dist.py::test_dist_mul[Levy]
tests/test_dist.py::test_dist_mul[Levy]
tests/test_dist.py::test_dist_mul[Levy]
tests/test_dist.py::test_dist_mul[LogGamma]
tests/test_dist.py::test_dist_mul[LogGamma]
tests/test_dist.py::test_dist_mul[LogGamma]
tests/test_dist.py::test_dist_mul[LogGamma]
tests/test_dist.py::test_dist_mul[LogGamma]
tests/test_dist.py::test_dist_mul[LogGamma]
tests/test_dist.py::test_dist_mul[LogLaplace]
tests/test_dist.py::test_dist_mul[LogLaplace]
tests/test_dist.py::test_dist_mul[LogLaplace]
tests/test_dist.py::test_dist_mul[LogLaplace]
tests/test_dist.py::test_dist_mul[LogLaplace]
tests/test_dist.py::test_dist_mul[LogLaplace]
tests/test_dist.py::test_dist_mul[LogWeibull]
tests/test_dist.py::test_dist_mul[LogWeibull]
tests/test_dist.py::test_dist_mul[LogWeibull]
tests/test_dist.py::test_dist_mul[LogWeibull]
tests/test_dist.py::test_dist_mul[LogWeibull]
tests/test_dist.py::test_dist_mul[LogWeibull]
tests/test_dist.py::test_dist_mul[Logistic]
tests/test_dist.py::test_dist_mul[Logistic]
tests/test_dist.py::test_dist_mul[Logistic]
tests/test_dist.py::test_dist_mul[Logistic]
tests/test_dist.py::test_dist_mul[Logistic]
tests/test_dist.py::test_dist_mul[Logistic]
tests/test_dist.py::test_dist_mul[Mielke]
tests/test_dist.py::test_dist_mul[Mielke]
tests/test_dist.py::test_dist_mul[Mielke]
tests/test_dist.py::test_dist_mul[Mielke]
tests/test_dist.py::test_dist_mul[Mielke]
tests/test_dist.py::test_dist_mul[Mielke]
tests/test_dist.py::test_dist_mul[Nakagami]
tests/test_dist.py::test_dist_mul[Nakagami]
tests/test_dist.py::test_dist_mul[Nakagami]
tests/test_dist.py::test_dist_mul[Nakagami]
tests/test_dist.py::test_dist_mul[Nakagami]
tests/test_dist.py::test_dist_mul[Nakagami]
tests/test_dist.py::test_dist_mul[Pareto1]
tests/test_dist.py::test_dist_mul[Pareto1]
tests/test_dist.py::test_dist_mul[Pareto1]
tests/test_dist.py::test_dist_mul[Pareto1]
tests/test_dist.py::test_dist_mul[Pareto1]
tests/test_dist.py::test_dist_mul[Pareto1]
tests/test_dist.py::test_dist_mul[Pareto2]
tests/test_dist.py::test_dist_mul[Pareto2]
tests/test_dist.py::test_dist_mul[Pareto2]
tests/test_dist.py::test_dist_mul[Pareto2]
tests/test_dist.py::test_dist_mul[Pareto2]
tests/test_dist.py::test_dist_mul[Pareto2]
tests/test_dist.py::test_dist_mul[PowerLogNormal]
tests/test_dist.py::test_dist_mul[PowerLogNormal]
tests/test_dist.py::test_dist_mul[PowerLogNormal]
tests/test_dist.py::test_dist_mul[PowerLogNormal]
tests/test_dist.py::test_dist_mul[PowerLogNormal]
tests/test_dist.py::test_dist_mul[PowerLogNormal]
tests/test_dist.py::test_dist_mul[PowerNormal]
tests/test_dist.py::test_dist_mul[PowerNormal]
tests/test_dist.py::test_dist_mul[PowerNormal]
tests/test_dist.py::test_dist_mul[PowerNormal]
tests/test_dist.py::test_dist_mul[PowerNormal]
tests/test_dist.py::test_dist_mul[PowerNormal]
tests/test_dist.py::test_dist_mul[TruncExponential]
tests/test_dist.py::test_dist_mul[TruncExponential]
tests/test_dist.py::test_dist_mul[TruncExponential]
tests/test_dist.py::test_dist_mul[TruncExponential]
tests/test_dist.py::test_dist_mul[TruncExponential]
tests/test_dist.py::test_dist_mul[TruncExponential]
tests/test_dist.py::test_dist_mul[TruncNormal]
tests/test_dist.py::test_dist_mul[TruncNormal]
tests/test_dist.py::test_dist_mul[TruncNormal]
tests/test_dist.py::test_dist_mul[TruncNormal]
tests/test_dist.py::test_dist_mul[TruncNormal]
tests/test_dist.py::test_dist_mul[TruncNormal]
tests/test_dist.py::test_dist_mul[TukeyLambda]
tests/test_dist.py::test_dist_mul[TukeyLambda]
tests/test_dist.py::test_dist_mul[TukeyLambda]
tests/test_dist.py::test_dist_mul[TukeyLambda]
tests/test_dist.py::test_dist_mul[TukeyLambda]
tests/test_dist.py::test_dist_mul[TukeyLambda]
tests/test_dist.py::test_dist_mul[Wald]
tests/test_dist.py::test_dist_mul[Wald]
tests/test_dist.py::test_dist_mul[Wald]
tests/test_dist.py::test_dist_mul[Wald]
tests/test_dist.py::test_dist_mul[Wald]
tests/test_dist.py::test_dist_mul[Wald]
tests/test_dist.py::test_dist_mul[WrappedCauchy]
tests/test_dist.py::test_dist_mul[WrappedCauchy]
tests/test_dist.py::test_dist_mul[WrappedCauchy]
tests/test_dist.py::test_dist_mul[WrappedCauchy]
tests/test_dist.py::test_dist_mul[WrappedCauchy]
tests/test_dist.py::test_dist_mul[WrappedCauchy]
tests/test_stress.py::test_quadrature
tests/test_stress.py::test_quadrature
tests/test_stress.py::test_quadrature
tests/test_stress.py::test_quadrature
tests/test_stress.py::test_approx_quadrature
tests/test_stress.py::test_integration
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/quad/collection/frontend.py:77: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
    parameters_spec = inspect.getargspec(quad_function)[0]
tests/test_dist.py::test_dist_add[ChiSquared]
tests/test_dist.py::test_dist_sub[ChiSquared]
tests/test_dist.py::test_dist_mul[ChiSquared]
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/collection/chi_squared.py:18: RuntimeWarning: divide by zero encountered in true_divide
    output *= (x/nc)**(0.25*df-0.5)
tests/test_dist.py::test_dist_add[ChiSquared]
tests/test_dist.py::test_dist_sub[ChiSquared]
tests/test_dist.py::test_dist_mul[ChiSquared]
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/collection/chi_squared.py:19: RuntimeWarning: invalid value encountered in multiply
    output *= special.iv(0.5*df-1, (nc*x)**0.5)
tests/test_dist.py::test_dist_add[GeneralizedExponential]
tests/test_dist.py::test_dist_add[Wald]
tests/test_dist.py::test_dist_sub[GeneralizedExponential]
tests/test_dist.py::test_dist_sub[Wald]
tests/test_dist.py::test_dist_mul[GeneralizedExponential]
tests/test_dist.py::test_dist_mul[Wald]
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/quad/collection/frontend.py:98: RuntimeWarning: invalid value encountered in true_divide
    weights /= numpy.sum(weights)
tests/test_dist.py::test_dist_add[GeneralizedExtreme]
tests/test_dist.py::test_dist_sub[GeneralizedExtreme]
tests/test_dist.py::test_dist_mul[GeneralizedExtreme]
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/collection/generalized_extreme.py:31: RuntimeWarning: invalid value encountered in true_divide
    return numpy.where(c == 0, x, -numpy.expm1(-c*x)/c)
tests/test_dist.py::test_dist_add[GeneralizedExtreme]
BUILDSTDERR: tests/test_dist.py::test_dist_sub[Generror: Bad exit status from /var/tmp/rpm-tmp.cIvVRK (%check)
eralizedExtreme]
tests/test_dist.py::test_dist_mul[GeneralizedExtreme]
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/collection/generalized_extreme.py:31: RuntimeWarning: invalid value encountered in double_scalars
    return numpy.where(c == 0, x, -numpy.expm1(-c*x)/c)
tests/test_dist.py::test_dist_add[GeneralizedExtreme]
tests/test_dist.py::test_dist_sub[GeneralizedExtreme]
tests/test_dist.py::test_dist_mul[GeneralizedExtreme]
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/collection/generalized_extreme.py:19: RuntimeWarning: invalid value encountered in true_divide
    logpex2 = numpy.where(c == 0, -x, logex2/c)
tests/test_dist.py::test_dist_add[RaisedCosine]
tests/test_dist.py::test_dist_add[RaisedCosine]
tests/test_dist.py::test_dist_add[RaisedCosine]
tests/test_dist.py::test_dist_sub[RaisedCosine]
tests/test_dist.py::test_dist_sub[RaisedCosine]
tests/test_dist.py::test_dist_sub[RaisedCosine]
tests/test_dist.py::test_dist_mul[RaisedCosine]
tests/test_dist.py::test_dist_mul[RaisedCosine]
tests/test_dist.py::test_dist_mul[RaisedCosine]
tests/test_dist.py::test_dist_mul[RaisedCosine]
tests/test_dist.py::test_dist_mul[RaisedCosine]
tests/test_dist.py::test_dist_mul[RaisedCosine]
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/collection/raised_cosine.py:27: DeprecationWarning: `hyp1f2` is deprecated!
  hyp1f2 is deprecated in SciPy 1.2
    for k_ in k.flatten()]).reshape(k.shape)
tests/test_dist.py::test_dist_add[Wald]
tests/test_dist.py::test_dist_sub[Wald]
tests/test_dist.py::test_dist_mul[Wald]
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/collection/wald.py:29: RuntimeWarning: invalid value encountered in multiply
    out -= numpy.exp(2.0/mu)*special.ndtr(-isqx*trm2)
tests/test_stress.py::test_approx_quadrature
  /builddir/build/BUILD/chaospy-3.0.5/chaospy/distributions/approximation.py:371: RuntimeWarning: invalid value encountered in true_divide
    floc = numpy.abs(floc / eps)
-- Docs: https://docs.pytest.org/en/latest/warnings.html
============= 4 failed, 185 passed, 430 warnings in 30.12 seconds ==============
```

Any hints on fixes/workarounds? Cheers.

**Comment by sanjayankur31** [comment]
Full log attached:

[build-log.txt](https://github.com/jonathf/chaospy/files/3457795/build-log.txt)

**Comment by jonathf** [comment]
Yes, I am aware. They should be fixed in 3.0.6. It has been on PyPI a week now.
I just pushed the git tag as well.

**Comment by sanjayankur31** [comment]
Ah! Thanks for that! The new version works well. Closing this now.
